providers/oidc.go

```diff
@@ -18,12 +24,19 @@ type OIDCProvider struct {
 	*ProviderData
 
 	SkipNonce bool
+
+	UseAssertionAuthentication bool
```

Comment:
I know this was part of my example in slack. I would prefer that we change this to an enum for better extensibility:

 ```suggestion
 type AuthenticationMethod int64
 
 const (
  // (default) https://oauth.net/2/client-authentication/
	ClientSecret AuthenticationMethod = iota
	// https://oauth.net/2/mtls/
	MutalTLS   
	// https://oauth.net/private-key-jwt/
	PrivateKeyJWT
)
 
	AuthenticationMethod AuthenticationMethod
```

Reply:
MTLS is currently implemented as part of https://github.com/oauth2-proxy/oauth2-proxy/pull/1838